config/routes.rb

```diff
@@ -5,35 +5,35 @@
   devise_for :users
   # get 'users/index'
   # root 'admin/plans#index'
-  root to: "users#index"
-  get "buy", to: "users#buy"
-  get "all_users", to: "users#all_users"
-  get "/search", to:"users#search"
-  #post "search", to: "users#search"
-  namespace :admin  do
-    resources :books do 
-      collection do 
+  root to: 'users#index'
+  get 'buy', to: 'users#buy'
+  get 'all_users', to: 'users#all_users'
+  get '/search', to: 'users#search'
+  # post "search", to: "users#search"
+  namespace :admin do
+    resources :books do
+      collection do
         get 'search'
       end
-    end 
+    end
     resources :plans
   end
 
   resources :issues
   resources :users
```

Comment:
why are you not using simpler way to eliminate unneeded routes is by specifying the :only